### DISCUSSION TITLE
GITHUB_ACTION environment variable missing - breaking issue

### DISCUSSION THREAD

**Discussion by davidjdixon** [community/community#25905] [Actions]
<p>Could someone from Github please confirm what has happened to the GITHUB_ACTION environment variable that is still documented, but is currently not being set inside workflows.</p>
<p>This is breaking a significant amount of our workflows in addition to a wide range of marketplace actions that depend on this documented environment variable being defined.</p>

**Comment by hross** [comment, marked as answer]
<p>There was a bug in a recent runner rollout: <a href="https://github.com/actions/runner/releases/tag/v2.272.0">https://github.com/actions/runner/releases/tag/v2.272.0</a>. We progressively roll runner updates to customers so some customers were impacted by this update (it broke GITHUB_ACTION context, as you saw). We are rolling it back now and all customers should be fixed by 10 am EST.</p>
<p>We apologize for the inconvenience and the issue here and have added some validation to the runner to make sure we have future coverage if the context is impacted before it rolls out to customers.</p>

**Comment by davidjdixon** [comment]
<p>Thanks for the update <a class="mention" href="/hross">@hross</a>, and appreciate the openness about the issue and your future mitigations</p>

**Comment by JGJP** [comment]
<p>I would like to say that I don’t think that this issue was handled well, and Github should rethink how it rolls out updates like these. Besides the sheer magnitude of this bug (completely breaking widely used actions like Reviewdog) that shouldn’t have made it into production in the first place, it seems that it was rolled out at the end of a workday, based on the timing of it and none of the staff being around to deal with the fallout. In Japan this meant an entire day of our workflows being broken, and it was not even clear what the issue was. We lost man hours on this and it was sheer luck that we didn’t need to rely on Actions for a hotfix.</p>

**Comment by donaldpiret** [comment]
<p><a class="mention" href="/hross">@hross</a> this seems to have broken for me again. Was the bad tag rolled out again without a proper fix?</p>

**Comment by hross** [comment]
<p>Can you give more detail on the issue you are seeing? We do not have any current issues I am aware of (and have not updated the runner recently). You can watch releases here: <a href="https://github.com/actions/runner/releases">https://github.com/actions/runner/releases</a>.</p>
<p>If you can reply to this comment with your repo and/or a link to a run we can investigate further.</p>

**Comment by hross** [comment]
<p>Thanks for keeping us honest here and we realize this caused unacceptable customer impact. We are using this as a chance to learn and update how we handle this type of issue. Once again I apologize for the breakage this caused and we hope our updated mechanisms for testing this and rolling it out will not cause further problems.</p>
